src/pages/Editor.vue

```diff
@@ -208,6 +216,13 @@
       @confirm="dialogConfirmed"
       @cancel="dialogCancelled"
     ></dialog-box>
+    <ImageUploaderDialog
+      v-if="showImageUploaderDialog"
+      :existingImage="itemImage"
+      @close-dialog="toggleImageUploaderBox"
+      @image-selected="uploadImage"
+      @delete-and-unlink-image="deleteLinkedImage"
```

Comment:
`delete-image` is enough? deleting will also unlink but that does not need to be explicitly told?